### PR TITLE
change for KanidmClientConfig to make it public

### PR DESCRIPTION
**Pull request by yaleman**
I wanna use this in [kanidm-profiles](https://github.com/yaleman/kanidm-profiles), and it should be the canonical definition for a Kanidm client config file 😄 

Also, minor 🖇️-calming.

- [x] cargo fmt has been run
- [x] cargo clippy has been run
- [x] cargo test has been run and passes

### CODE REVIEW

File: kanidm_client/src/lib.rs [hunk 1]
```diff
@@ -61,13 +61,15 @@ pub enum ClientError {
...
 #[derive(Debug, Deserialize)]
-struct KanidmClientConfig {
+pub struct KanidmClientConfig {
     uri: Option<String>,
     verify_ca: Option<bool>,
     verify_hostnames: Option<bool>,
     ca_path: Option<String>,
-    // Should we add username/pw later? They could be part of the builder
-    // process ...
+    #[allow(dead_code)]
```
Comment (Firstyear) [inline kanidm_client/src/lib.rs:69]: I think we just get rid of these two fields for now since we don't use them today.

**Review by Firstyear — APPROVED** [review]
Minor comment, but looks good.

File: kanidm_client/src/lib.rs (hunk 1 above)
Comment (yaleman) [reply kanidm_client/src/lib.rs:69]: Maybe comment them out? I was going for 'the canonical definition for a Kanidm client config file' to try to keep `pykanidm` and the core def together. Happy either way.

Comment (Firstyear) [reply kanidm_client/src/lib.rs:69]: We don't use them, so remove them and we can readd if needed

**Review by Firstyear — APPROVED** [review]

**Pull request merged**